extension/chrome/elements/compose-modules/compose-draft-module.ts

```diff
@@ -113,12 +113,25 @@ export class ComposeDraftModule extends ViewModule<ComposeView> {
   };
 
   public draftSave = async (forceSave: boolean = false): Promise<void> => {
+    if (this.disableSendingDrafts) {
+      return;
+    }
     if (this.hasBodyChanged(this.view.inputModule.squire.getHTML()) || this.hasSubjectChanged(String(this.view.S.cached('input_subject').val())) || forceSave) {
       this.currentlySavingDraft = true;
       try {
         const msgData = this.view.inputModule.extractAll();
-        const primaryKi = await this.view.storageModule.getKey(msgData.from);
-        const pubkeys = [{ isMine: true, email: msgData.from, pubkey: await KeyUtil.parse(primaryKi.public) }];
+        const { pubkeys } = await this.view.storageModule.collectSingleFamilyKeys([], msgData.from, true);
+        // collectSingleFamilyKeys filters out bad keys, but only if there are any good keys available
+        //  if no good keys available, it leaves bad keys so we can explain the issue here
+        if (pubkeys.find(pub => pub.pubkey.expiration && pub.pubkey.expiration > Date.now())) {
+          throw new UnreportableError('Your account keys are expired');
+        }
+        if (pubkeys.find(pub => pub.pubkey.revoked)) {
+          throw new UnreportableError('Your account keys are revoked');
+        }
+        if (pubkeys.find(pub => !pub.pubkey.usableForEncryption)) {
+          throw new UnreportableError('Your account keys are not usable for encryption');
+        }
```

Comment:
It may be interesting if these could move into `collectSingleFamilyKeysInternal`. That would make it less messy, plus same error messages would apply both to signing and sending. I may have a look in another issue.